src/ast/passes/macro_expansion.cpp

```diff
@@ -416,15 +416,25 @@ std::optional<BlockExpr *> MacroExpander::expand(const Macro &macro,
   return cloned_block;
 }
 
+void expand_macro(ASTContext &ast,
+                  Expression &expr,
+                  const MacroRegistry &registry)
+{
+  std::vector<const Macro *> stack;
+  MacroExpander expander(ast, registry, stack);
+  expander.visit(expr);
+}
+
 Pass CreateMacroExpansionPass()
 {
-  auto fn = [](ASTContext &ast) {
+  auto fn = [](ASTContext &ast) -> MacroRegistry {
     auto macros = MacroRegistry::create(ast);
     if (ast.diagnostics().ok()) {
```

Comment:
Nit: I'm not sure why this is necessary, but the pass won't be run if the diagnostics are already bad.

Reply:
Good call. Removed.